### PR TITLE
Feature/serializer

### PR DESCRIPTION
**Pull request by d-srd**

### CODE REVIEW

File: Sources/jab/Identifiable.swift [hunk 1]
```diff
@@ -10,11 +10,5 @@ import Foundation
...
     var identifier: String { get set }
     
-    static var codingKey: String { get }
-}
-
-extension JSONAPIIdentifiable {
-    static var codingKey: String {
-        return "id"
-    }
+    static var jsonTypeIdentifier: String { get }
```
Comment (fajdof) [inline Sources/jab/Identifiable.swift:13]: what is the difference between jsonType and plain identifier?

File: Sources/jab/JSONAPISerializer.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        var relationships = Dictionary<String, Any>()
+        
+        reflectNamedProperties(of: resource) { (child: Resource, label) in
+            let relation: [String: [String: Any]] = [
+                JSONAPIKeys.data.rawValue: [
+                    JSONAPIKeys.type.rawValue: type(of: child).jsonTypeIdentifier,
+                    JSONAPIKeys.id.rawValue: child.identifier
+                ]
+            ]
+            relationships[label] = relation
```
Comment (fajdof) [inline Sources/jab/JSONAPISerializer.swift:44]: what is label exactly?

File: Sources/jab/JSONAPISerializer.swift [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+//        return Result(catching: { try serialize(resource: resource) })
+//    }
+//
+//    public func deserializeCollection<T: Resource>(data: Data) throws -> Paginated<T> {
+//        let jsonDataOption = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? Dictionary<String, Any>
+//        let jsonData = try unwrap(jsonDataOption, orThrow: Error.notConvertibleToDictionary(data: data))
+//        
+//        let linksDict = try unwrap(jsonData[JSONAPIKeys.links] as? Dictionary<String, Any>, orThrow: Error.missingLinksParameter(dictionary: jsonData))
+//        let linksData = try JSONSerialization.data(withJSONObject: linksDict, options: .prettyPrinted)
+//        let links = try decoder.decode(Links.self, from: linksData)
```
Comment (fajdof) [inline Sources/jab/JSONAPISerializer.swift]: can these be deleted?

File: Sources/jab/JSONAPISerializer.swift [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        let mirror = Mirror(reflecting: target)
+        
+        for child in mirror.children {
+            if let value = child.value as? T, let label = child.label {
+                closure(value, label)
+            }
+            
+            if recursively {
+                // To enable recursive reflection, all we have to do
+                // is to call our own method again, using the value
```
Comment (fajdof) [inline Sources/jab/JSONAPISerializer.swift]: this comment is great

File: Sources/jab/Identifiable.swift (hunk 1 above)
Comment (d-srd) [reply Sources/jab/Identifiable.swift:13]: oba su dio json:api specifikacije - https://jsonapi.org/format/#document-resource-object-identification

identifier je, uobicajeno, neki UUID string
jsonTypeIdentifier je neki string koji uniquely identifiea tip objekta

File: Sources/jab/JSONAPISerializer.swift (hunk 3 above)
Comment (d-srd) [reply Sources/jab/JSONAPISerializer.swift]: maybe

File: Sources/jab/JSONAPISerializer.swift (hunk 2 above)
Comment (d-srd) [reply Sources/jab/JSONAPISerializer.swift:44]: pogledaj definiciju funkcije `reflectNamedProperties` i kako ona poziva ovaj closure

**Review by fajdof — APPROVED** [review]

**Pull request merged**